identity/app/services/AuthenticationService.scala

```diff
@@ -25,7 +25,9 @@ class AuthenticationService(cookieDecoder: FrontendIdentityCookieDecoder,
     minimalSecureUser <- cookieDecoder.getUserDataForScGuU(scGuU.value)
     guUCookieData <- cookieDecoder.getUserDataForGuU(guU.value)
     fullUser = guUCookieData.getUser if (fullUser.getId == minimalSecureUser.getId)
-  } yield AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData))
+    scGuLa <- request.cookies.get("SC_GU_LA")
```

Comment:
So in the case where a user has all valid cookies, but is missing "SC_GU_LA" then this for comprehension returns `None`?  I don't think that's the intention?

Reply:
ah you're right, something like  : 
```
 def authenticatedUserFor[A](request: RequestHeader): Option[AuthenticatedUser] = for {
    scGuU <- request.cookies.get("SC_GU_U")
    guU <- request.cookies.get("GU_U")
    minimalSecureUser <- cookieDecoder.getUserDataForScGuU(scGuU.value)
    guUCookieData <- cookieDecoder.getUserDataForGuU(guU.value)
    fullUser = guUCookieData.getUser if (fullUser.getId == minimalSecureUser.getId)
  } yield {
    val hasRecentlyAuthenticated = request.cookies.get("SC_GU_LA").map(scgula => cookieDecoder.userHasRecentScGuLaCookie(fullUser,scgula.value, Minutes.minutes(20).toStandardDuration))
    AuthenticatedUser(fullUser, ScGuU(scGuU.value, guUCookieData), hasRecentlyAuthenticated.getOrElse(false))
  }
```
??